Unable to decode JWK response from Azure JWK

Current jwt authenticator fails with following stack trace if set `http-server.authentication.jwt.key-file` to `https://login.microsoftonline.com/TENANT_ID/discovery/v2.0/keys`

Relevant Discussion on slack: https://prestosql.slack.com/archives/CGGL0EZ9N/p1607500620014400

```
Caused by: java.lang.RuntimeException: Unable to decode JWK response from https://login.microsoftonline.com/TENANT_ID/discovery/v2.0/keys
  at io.prestosql.server.security.jwt.JwkService.fetchKeys(JwkService.java:149)
  at io.prestosql.server.security.jwt.JwkService.<init>(JwkService.java:69)
  at io.prestosql.server.security.jwt.JwkService.<init>(JwkService.java:60)
  at io.prestosql.server.security.jwt.JwkService$$FastClassByGuice$$13916454.newInstance(<generated>)
  at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
  at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
  at com.google.inject.internal.ConstructorInjector.access$000(ConstructorInjector.java:32)
  at com.google.inject.internal.ConstructorInjector$1.call(ConstructorInjector.java:98)
  at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:112)
  at io.airlift.bootstrap.LifeCycleModule.provision(LifeCycleModule.java:54)
  at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:120)
  at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
  at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:93)
  at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
  at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
  at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
  at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
  at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
  at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
  at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
  at com.google.inject.internal.ConstructorInjector.access$000(ConstructorInjector.java:32)
  at com.google.inject.internal.ConstructorInjector$1.call(ConstructorInjector.java:98)
  at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:112)
  at io.airlift.bootstrap.LifeCycleModule.provision(LifeCycleModule.java:54)
  at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:120)
  at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
  at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:93)
  at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
  at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
  at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
  at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
  at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
  at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
  at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
  at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
  at com.google.inject.internal.ConstructorInjector.access$000(ConstructorInjector.java:32)
  at com.google.inject.internal.ConstructorInjector$1.call(ConstructorInjector.java:98)
  at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:112)
  at io.airlift.bootstrap.LifeCycleModule.provision(LifeCycleModule.java:54)
  at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:120)
  at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
  at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:93)
  at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
  at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
  at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
  at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
  at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
  at com.google.inject.internal.RealOptionalBinder$JavaOptionalProvider.doProvision(RealOptionalBinder.java:279)
  at com.google.inject.internal.RealOptionalBinder$JavaOptionalProvider.doProvision(RealOptionalBinder.java:240)
  at com.google.inject.internal.InternalProviderInstanceBindingImpl$Factory$1.call(InternalProviderInstanceBindingImpl.java:120)
  at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:112)
  at io.airlift.bootstrap.LifeCycleModule.provision(LifeCycleModule.java:54)
  at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:120)
  at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
  at com.google.inject.internal.InternalProviderInstanceBindingImpl$Factory.get(InternalProviderInstanceBindingImpl.java:115)
  at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
  at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
  at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
  at com.google.inject.internal.ConstructorInjector.access$000(ConstructorInjector.java:32)
  at com.google.inject.internal.ConstructorInjector$1.call(ConstructorInjector.java:98)
  at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:112)
  at io.airlift.bootstrap.LifeCycleModule.provision(LifeCycleModule.java:54)
  at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:120)
  at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
  at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:93)
  at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
  at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
  at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
  at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
  at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
  at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
  at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
  at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
  at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:213)
  at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:184)
  at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:111)
  at com.google.inject.Guice.createInjector(Guice.java:87)
  at io.airlift.bootstrap.Bootstrap.initialize(Bootstrap.java:276)
  at io.prestosql.server.Server.doStart(Server.java:111)
  at io.prestosql.server.Server.lambda$start$0(Server.java:73)
  at io.prestosql.$gen.Presto_345____20201210_072203_1.run(Unknown Source)
  at io.prestosql.server.Server.start(Server.java:73)
  at io.prestosql.server.PrestoServer.main(PrestoServer.java:38)
Caused by: java.lang.IllegalArgumentException: Invalid JSON string for [simple type, class io.prestosql.server.security.jwt.JwkDecoder$Keys]
  at io.airlift.json.JsonCodec.fromJson(JsonCodec.java:137)
  at io.prestosql.server.security.jwt.JwkDecoder.decodeKeys(JwkDecoder.java:47)
  at io.prestosql.server.security.jwt.JwkService.fetchKeys(JwkService.java:146)
  ... 82 more
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize instance of `java.lang.String` out of START_ARRAY token
 at [Source: (String)"{"keys":[{"kty":"RSA","use":"sig","kid":"kg2LYs2T0CTjIfj4rt6JIynen38","x5t":"kg2LYs2T0CTjIfj4rt6JIynen38","n":"yTKa6m5GFOllz7oIHFCkvRJoBv7wLMuKIPLHbFGh5yOiO8o3akoqMhf1x6MxINGhZo6dkIrhVlVfWJhEJZPVaQdvyvVmlIZruhcbz3PGMqPAbjq2JqbB1mMnsyGHx-ovP0Cm5xj8sgI8wm67p3nosqzqFvg6mPKVO-w1QBr5seDU2AwU2DR88LF2v03Zjgn4mGvPdUOXihTQoNlf-nJFduXMDyRgZabnR2HlYHhagHwy1beWW1WtEaPz8iBN_0bGkGw705aDBUHJkdTty1mzsCZRur_n0imqXu9IzoSyiq5d0yKrRA5xkA-K3DMeRMquZ5QvPT9Eee4EZfFL97zBfQ","e":"AQAB","x5c":["MIIDBTCCAe2gAwIBAgIQQiR8gZ"[truncated 4262 chars]; line: 1, column: 473] (through reference chain: io.prestosql.server.security.jwt.JwkDecoder$Keys["keys"]->java.util.ArrayList[0]->java.util.LinkedHashMap["x5c"])
  at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
  at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1445)
  at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1219)
  at com.fasterxml.jackson.databind.deser.std.StdDeserializer._deserializeFromArray(StdDeserializer.java:714)
  at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:40)
  at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:10)
  at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap(MapDeserializer.java:527)
  at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:364)
  at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:29)
  at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:286)
  at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245)
  at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:27)
  at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:530)
  at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:528)
  at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:417)
  at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1292)
  at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:326)
  at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159)
  at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:1719)
  at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1261)
  at io.airlift.json.JsonCodec.fromJson(JsonCodec.java:134)
  ... 84 more
```

--- a/core/trino-main/src/main/java/io/trino/server/security/jwt/JwkDecoder.java
+++ b/core/trino-main/src/main/java/io/trino/server/security/jwt/JwkDecoder.java
@@ -13,6 +13,7 @@
  */
 package io.trino.server.security.jwt;
 
+import com.fasterxml.jackson.annotation.JsonAnySetter;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.base.Strings;
@@ -26,6 +27,7 @@ import java.security.interfaces.ECPublicKey;
 import java.security.interfaces.RSAPublicKey;
 import java.security.spec.ECParameterSpec;
 import java.security.spec.ECPoint;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
@@ -52,84 +54,54 @@ public final class JwkDecoder
                 .collect(toImmutableMap(JwkPublicKey::getKeyId, Function.identity()));
     }
 
-    public static Optional<? extends JwkPublicKey> tryDecodeJwkKey(Map<String, String> properties)
+    public static Optional<? extends JwkPublicKey> tryDecodeJwkKey(Key key)
     {
-        String keyId = properties.get("kid");
-        if (Strings.isNullOrEmpty(keyId)) {
-            // key id is required to index the key
+        // key id is required to index the key
+        if (key.getKid().isEmpty() || key.getKid().get().isEmpty()) {
             return Optional.empty();
         }
-
-        String keyType = properties.get("kty");
-        switch (keyType) {
+        String keyId = key.getKid().get();
+        switch (key.getKty()) {
             case "RSA":
-                return tryDecodeRsaKey(keyId, properties);
+                return tryDecodeRsaKey(keyId, key);
             case "EC":
-                return tryDecodeEcKey(keyId, properties);
+                return tryDecodeEcKey(keyId, key);
             default:
                 // ignore non unknown keys
                 return Optional.empty();
         }
     }
 
-    public static Optional<JwkRsaPublicKey> tryDecodeRsaKey(String keyId, Map<String, String> properties)
+    public static Optional<JwkRsaPublicKey> tryDecodeRsaKey(String keyId, Key key)
     {
         // alg field is optional so not verified
         // use field is optional so not verified
-
-        String encodedModulus = properties.get("n");
-        if (Strings.isNullOrEmpty(encodedModulus)) {
-            log.error("JWK RSA key %s does not contain the required modulus field 'n'", keyId);
-            return Optional.empty();
-        }
-        String encodedExponent = properties.get("e");
-        if (Strings.isNullOrEmpty(encodedExponent)) {
-            log.error("JWK RSA key %s does not contain the required exponent field 'e'", keyId);
-            return Optional.empty();
-        }
-
-        Optional<BigInteger> modulus = decodeBigint(keyId, "modulus", encodedModulus);
+        Optional<BigInteger> modulus = key.getStringProperty("n").flatMap(encodedModulus -> decodeBigint(keyId, "modulus", encodedModulus));
         if (modulus.isEmpty()) {
             return Optional.empty();
         }
-        Optional<BigInteger> exponent = decodeBigint(keyId, "exponent", encodedExponent);
+        Optional<BigInteger> exponent = key.getStringProperty("e").flatMap(encodedExponent -> decodeBigint(keyId, "exponent", encodedExponent));
         if (exponent.isEmpty()) {
             return Optional.empty();
         }
         return Optional.of(new JwkRsaPublicKey(keyId, exponent.get(), modulus.get()));
     }
 
-    public static Optional<JwkEcPublicKey> tryDecodeEcKey(String keyId, Map<String, String> properties)
+    public static Optional<JwkEcPublicKey> tryDecodeEcKey(String keyId, Key key)
     {
         // alg field is optional so not verified
         // use field is optional so not verified
-
-        String curveName = properties.get("crv");
-        if (Strings.isNullOrEmpty(curveName)) {
-            log.error("JWK EC key %s does not contain the required curve field 'crv'", keyId);
-            return Optional.empty();
-        }
-        String encodedX = properties.get("x");
-        if (Strings.isNullOrEmpty(encodedX)) {
-            log.error("JWK EC key %s does not contain the required x coordinate field 'x'", keyId);
-            return Optional.empty();
-        }
-        String encodedY = properties.get("y");
-        if (Strings.isNullOrEmpty(encodedY)) {
-            log.error("JWK EC key %s does not contain the required y coordinate field 'y'", keyId);
-            return Optional.empty();
-        }
-
-        Optional<ECParameterSpec> curve = EcCurve.tryGet(curveName);
+        Optional<String> curveName = key.getStringProperty("crv");
+        Optional<ECParameterSpec> curve = curveName.flatMap(EcCurve::tryGet);
         if (curve.isEmpty()) {
             log.error("JWK EC %s curve '%s' is not supported", keyId, curveName);
             return Optional.empty();
         }
-        Optional<BigInteger> x = decodeBigint(keyId, "x", encodedX);
+        Optional<BigInteger> x = key.getStringProperty("x").flatMap(encodedX -> decodeBigint(keyId, "x", encodedX));
         if (x.isEmpty()) {
             return Optional.empty();
         }
-        Optional<BigInteger> y = decodeBigint(keyId, "y", encodedY);
+        Optional<BigInteger> y = key.getStringProperty("y").flatMap(encodedY -> decodeBigint(keyId, "y", encodedY));
         if (y.isEmpty()) {
             return Optional.empty();
         }
@@ -259,17 +231,58 @@ public final class JwkDecoder
 
     public static class Keys
     {
-        private final List<Map<String, String>> keys;
+        private final List<Key> keys;
 
         @JsonCreator
-        public Keys(@JsonProperty("keys") List<Map<String, String>> keys)
+        public Keys(@JsonProperty("keys") List<Key> keys)
         {
             this.keys = ImmutableList.copyOf(requireNonNull(keys, "keys is null"));
         }
 
-        public List<Map<String, String>> getKeys()
+        public List<Key> getKeys()
         {
             return keys;
+        }
+    }
+
+    public static class Key
+    {
+        private final String kty;
+        private final Optional<String> kid;
+        private final Map<String, Object> other = new HashMap<>();
+
+        @JsonCreator
+        public Key(
+                @JsonProperty("kty") String kty,
+                @JsonProperty("kid") Optional<String> kid)
+        {
+            this.kty = requireNonNull(kty, "kty is null");
+            this.kid = requireNonNull(kid, "kid is null");
+        }
+
+        public String getKty()
+        {
+            return kty;
+        }
+
+        public Optional<String> getKid()
+        {
+            return kid;
+        }
+
+        public Optional<String> getStringProperty(String name)
+        {
+            Object value = other.get(name);
+            if (value instanceof String && !Strings.isNullOrEmpty((String) value)) {
+                return Optional.of((String) value);
+            }
+            return Optional.empty();
+        }
+
+        @JsonAnySetter
+        public void set(String name, Object value)
+        {
+            other.put(name, value);
         }
     }
 }

--- a/core/trino-main/src/test/java/io/trino/server/security/jwt/TestJwkService.java
+++ b/core/trino-main/src/test/java/io/trino/server/security/jwt/TestJwkService.java
@@ -56,6 +56,18 @@ public class TestJwkService
             "      \"x\": \"W9pnAHwUz81LldKjL3BzxO1iHe1Pc0fO6rHkrybVy6Y\",\n" +
             "      \"y\": \"XKSNmn_xajgOvWuAiJnWx5I46IwPVJJYPaEpsX3NPZg\",\n" +
             "      \"alg\": \"ES256\"\n" +
+            "    },\n" +
+            "    {\n" +
+            "      \"alg\": \"RS256\",\n" +
+            "      \"kty\": \"RSA\",\n" +
+            "      \"use\": \"sig\",\n" +
+            "      \"x5c\": [\n" +
+            "           \"MIIC+DCCAeCgAwIBAgIJBIGjYW6hFpn2MA0GCSqGSIb3DQEBBQUAMCMxITAfBgNVBAMTGGN1c3RvbWVyLWRlbW9zLmF1dGgwLmNvbTAeFw0xNjExMjIyMjIyMDVaFw0zMDA4MDEyMjIyMDVaMCMxITAfBgNVBAMTGGN1c3RvbWVyLWRlbW9zLmF1dGgwLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMnjZc5bm/eGIHq09N9HKHahM7Y31P0ul+A2wwP4lSpIwFrWHzxw88/7Dwk9QMc+orGXX95R6av4GF+Es/nG3uK45ooMVMa/hYCh0Mtx3gnSuoTavQEkLzCvSwTqVwzZ+5noukWVqJuMKNwjL77GNcPLY7Xy2/skMCT5bR8UoWaufooQvYq6SyPcRAU4BtdquZRiBT4U5f+4pwNTxSvey7ki50yc1tG49Per/0zA4O6Tlpv8x7Red6m1bCNHt7+Z5nSl3RX/QYyAEUX1a28VcYmR41Osy+o2OUCXYdUAphDaHo4/8rbKTJhlu8jEcc1KoMXAKjgaVZtG/v5ltx6AXY0CAwEAAaMvMC0wDAYDVR0TBAUwAwEB/zAdBgNVHQ4EFgQUQxFG602h1cG+pnyvJoy9pGJJoCswDQYJKoZIhvcNAQEFBQADggEBAGvtCbzGNBUJPLICth3mLsX0Z4z8T8iu4tyoiuAshP/Ry/ZBnFnXmhD8vwgMZ2lTgUWwlrvlgN+fAtYKnwFO2G3BOCFw96Nm8So9sjTda9CCZ3dhoH57F/hVMBB0K6xhklAc0b5ZxUpCIN92v/w+xZoz1XQBHe8ZbRHaP1HpRM4M7DJk2G5cgUCyu3UBvYS41sHvzrxQ3z7vIePRA4WF4bEkfX12gvny0RsPkrbVMXX1Rj9t6V7QXrbPYBAO+43JvDGYawxYVvLhz+BJ45x50GFQmHszfY3BR9TPK8xmMmQwtIvLu1PMttNCs7niCYkSiUv2sc2mlq1i3IashGkkgmo=\"\n" +
+            "      ],\n" +
+            "      \"n\": \"yeNlzlub94YgerT030codqEztjfU_S6X4DbDA_iVKkjAWtYfPHDzz_sPCT1Axz6isZdf3lHpq_gYX4Sz-cbe4rjmigxUxr-FgKHQy3HeCdK6hNq9ASQvMK9LBOpXDNn7mei6RZWom4wo3CMvvsY1w8tjtfLb-yQwJPltHxShZq5-ihC9irpLI9xEBTgG12q5lGIFPhTl_7inA1PFK97LuSLnTJzW0bj096v_TMDg7pOWm_zHtF53qbVsI0e3v5nmdKXdFf9BjIARRfVrbxVxiZHjU6zL6jY5QJdh1QCmENoejj_ytspMmGW7yMRxzUqgxcAqOBpVm0b-_mW3HoBdjQ\",\n" +
+            "      \"e\": \"AQAB\",\n" +
+            "      \"kid\": \"test-certificate-chain\",\n" +
+            "      \"x5t\": \"NjVBRjY5MDlCMUIwNzU4RTA2QzZFMDQ4QzQ2MDAyQjVDNjk1RTM2Qg\"\n" +
             "    }\n" +
             "  ]\n" +
             "}";
@@ -185,8 +197,9 @@ public class TestJwkService
     private static void assertTestKeys(JwkService service)
     {
         Map<String, PublicKey> keys = service.getKeys();
-        assertEquals(keys.size(), 2);
+        assertEquals(keys.size(), 3);
         assertTrue(keys.containsKey("test-rsa"));
         assertTrue(keys.containsKey("test-ec"));
+        assertTrue(keys.containsKey("test-certificate-chain"));
     }
 }

